t/repos.t

```diff
@@ -87,28 +88,59 @@ subtest "Pithub::Repos->delete" => sub {
     }
 }
 
+# Pithub::Repos::Branches->get
+{
+    my $obj = Pithub::Test::Factory->create( 'Pithub::Repos::Branches', user => 'foo', repo => 'bar' );
 
-subtest "Pithub::Repos->branch" => sub {
-    my $obj = Pithub::Test::Factory->create(
-        'Pithub::Repos',
-        user => 'foo',
-        repo => 'bar'
-    );
```

Comment:
Thanks for this @thibaultduponchelle! I'm just looking at the tests and wondering if we could add new tests without rewriting the old tests as that would serve to prove that the API is still backwards compatible.

Reply:
> Thanks for this @thibaultduponchelle! I'm just looking at the tests and wondering if we could add new tests without rewriting the old tests as that would serve to prove that the API is still backwards compatible.

Do you think it’s possible?? 🧐🤔

Reply:
I don't know, as I haven't looked too closely. I didn't see anything here that said it wasn't backwards compatible. 😉

Reply:
🙊 😁 

(sorry I should have claimed loudly since I was well aware of this)

I'm afraid it's a breaking change. Currently there is a `branch` and `branches` that are methods to call `get` and `list` methods. With this pull request, `branches` will give access to `Pithub::Repos::Branches`like it is done for `Pithub::Repos::Commits`.

I'm not sure how to continue to make `branches` answers both `list` and  `Pithub::Repos::Branches`.

My initial contribution is to add `rename` and `merge` methods, the refactoring seemed meaningful to me but have impacts you're right.

I you consider it is a "no way" then I could add 2 new methods to `Pithub::Repos` like branch_rename and `branch_merge`.

Tell me 😃

Reply:
My first thought would be to add the new methods, maybe `rename_branch` and `merge_branch`, as that would add the functionality without breaking the existing API. Not sure if there is already a pattern for something like this in the repo.

My thinking is that I just recently merged what felt like a small change in a CPAN module which then broke a different module and had to be debugged and rolled back etc. Anyway, was a bunch of extra work because I allowed a surprising change to be enabled by default. Would be nice to avoid that here. 😄

Reply:
I totally understand! Let’s do this then 👍🏼